R/responsibilities.R

```diff
@@ -64,20 +69,44 @@ montagu_touchstone_versions <- function(modelling_group_id,
   collect_data
 }
 
-helper_get_touchstone <- function(modelling_group_id, touchstone_id,
+################################################################################
+
+helper_get_touchstone <- function(modelling_group_id, 
+                                  touchstone_id,
                                   location = NULL) {
-  assert_scalar_character(modelling_group_id)
-  assert_scalar_character(touchstone_id)
-  path <- sprintf("/modelling-groups/%s/responsibilities/%s/",
-                  modelling_group_id, touchstone_id)
-  montagu_api_GET(location, path)
+  if (is.null(modelling_group_id)) {
+    path <- sprintf("/touchstones/%s/responsibilities/",
+                    touchstone_id)
+    montagu_api_GET(location, path)
+  
+  } else {
+    assert_scalar_character(modelling_group_id)
+    assert_scalar_character(touchstone_id)
+    path <- sprintf("/modelling-groups/%s/responsibilities/%s/",
+                    modelling_group_id, touchstone_id)
+    montagu_api_GET(location, path)
+  }
+}
+
+helper_get_responsibilities <- function(modelling_group_id, touchstone_id, 
+                                        location) {
+  resps <- helper_get_touchstone(modelling_group_id, touchstone_id, location)
+  
+  if (is.null(modelling_group_id)) {
+    resps <- lapply(resps, "[[", "responsibilities")[[1]]
```

Comment:
is this not equivalent to `resps[[1]]$responsibilities`, and is that what is wanted?

Reply:
Good spot - that's wrong, not sure what I was thinking. The results looked good because the modelling group at index 1 is VIMC, which had all the scenarios. The [[1]] shouldn't be there, and the responsibilities for all the groups should be rbind-ed, and unique-d in montagu_scenarios - which is the only place where this can be called with a NULL modelling_group_id

Reply:
Fixed and tested!